tests/rptest/redpanda_cloud_tests/omb_validation_test.py

```diff
@@ -289,7 +289,12 @@ def test_max_connections(self):
     def test_max_partitions(self):
         tier_limits = self.tier_limits
 
-        partitions_per_topic = tier_limits.max_partition_count
+        # Producer clients perform poorly with many partitions. Hence we limit
+        # the max amount per producer by splitting them over multiple topics.
+        MAX_PARTITIONS_PER_TOPIC = 5000
+        topics = math.ceil(tier_limits.max_partition_count / MAX_PARTITIONS_PER_TOPIC)
+
+        partitions_per_topic = tier_limits.max_partition_count // topics
```

Comment:
nit: I think you could do a `ciel(... / topics)` here as above instead of `//` so we are conservative with the rounding: if not exact we will test slightly more than the advertised number rather than slightly less.

Reply:
Yep makes sense, amended.